velox/exec/HashPartitionFunction.cpp

```diff
@@ -57,7 +57,12 @@ void HashPartitionFunction::init(
 std::optional<uint32_t> HashPartitionFunction::partition(
     const RowVector& input,
     std::vector<uint32_t>& partitions) {
-  auto size = input.size();
+  const auto size = input.size();
+
+  if (hashers_.empty() && !hashBitRange_.has_value()) {
```

Comment:
A better fix might be to change presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp to check if partitioning keys are empty and set numPartitions to 1. 

CC: @karteekmurthys

Reply:
+1.

Agree with @mbasmanova. Its better to add a check here instead of fix this way.

We could change PrestoToVeloxQueryPlan.cpp for empty partitioning keys. But we also should figure why the planner didn't add a broadcast exchange for the SQL query we had.

Reply:
Sure, I will update this PR to only check the validity of hash function creation once the planner part is fixed.